ZA: members' interests data 2015 and addition of source links

geoffkilpin opened the pull request:
2015 members' interests data ready for import. Also adds new sourceAFRICA informationsource links for releases from all years and adds links to the members' interests browser and person profiles.

Because of the update to data from previous years, all data needs to be reimported. Please run ```interests_register_delete_existing``` and then:

```
$ ./manage.py interests_register_import_from_json pombola/south_africa/data/members-interests/2010_for_import.json
$ ./manage.py interests_register_import_from_json pombola/south_africa/data/members-interests/2011_for_import.json
$ ./manage.py interests_register_import_from_json pombola/south_africa/data/members-interests/2012_for_import.json
$ ./manage.py interests_register_import_from_json pombola/south_africa/data/members-interests/2013_for_import.json
$ ./manage.py interests_register_import_from_json pombola/south_africa/data/members-interests/2014_for_import.json
$ ./manage.py interests_register_import_from_json pombola/south_africa/data/members-interests/2015_for_import.json
```

duncanparkes commented:
This looks pretty good to me - I have just a few comments on the individual commits.

When this is done on the live database I expect there to be some slugs of people that aren't matched. When I imported this data on my own dev environment the following person slugs were missing from my dataset:

'cathlene-labuschagne'}
'tandi-gloria-mpambo-sibhukwana'
'marshall-mzingisi-dlamini}
'zukiswa-veronica-ncitha'

At least some of these are missing on the live site too.

geoffkilpin commented:
Thanks for the useful comments @duncanparkes - I've updated the relevant commits.

Regarding the matching of slugs - thanks for picking up on those (I think it is a result of the issues with core_import_popolo in #1515 - which I used to replicate the live database). I've added corrections for all those you list, except for 'zukiswa-veronica-ncitha' - which [appears to be correct](http://www.pa.org.za/person/zukiswa-veronica-ncitha/). Seeing that I don't have an accurate replicated database, would you mind please re-testing the import before applying it to the live site?

mhl commented on pombola/south_africa/templates/interests_register/index_complete.html:
```diff
@@ -3,7 +3,11 @@
     <h3>
         <a href="{{ person.person.get_absolute_url }}">{{ person.person.name }}</a>
-         - {{ person.year }}
+         - {% if person.source_url == "" %}
```
A tiny thing, but In the three cases here where you've done `if foo == ""` here, it's probably clearer to do `if not foo` or reverse the `if` and `else` clauses and just do `if foo`, e.g.:

```
         - {% if person.source_url %}
             <a href="{{ person.source_url }}" target="_blank" title="View Source Document">{{ person.year }}</a>
           {% else %}
            {{ person.year }}
           {% endif %}
```

mhl commented:
I've tried this again on our staging site after replicating the live database to there.

The 2010 and 2012 imports fails to find:

mhl commented:
Hi @geoffkilpin - thanks very much for all this work. I've replicated the live database to our staging site and run the import there.  There are a few slugs for people that can't be found for 2014 and 2015, though:

```
$ for y in 2010 2011 2012 2013 2014 2015
> do
> echo Year $y
> ./manage.py interests_register_import_from_json pombola/south_africa/data/members-interests/${y}_for_import.json
> done
Year 2010
Year 2011
Year 2012
Year 2013
Year 2014
Failed to find the person from: {u'slug': u'cathlene-labuschagne'}
Failed to find the person from: {u'slug': u'cathlene-labuschagne'}
Failed to find the person from: {u'slug': u'hlengiwe-octavia-maxon'}
Failed to find the person from: {u'slug': u'tandi-gloria-mpambo-sibhukwana'}
Failed to find the person from: {u'slug': u'tandi-gloria-mpambo-sibhukwana'}
Year 2015
Failed to find the person from: {u'slug': u'cathlene-labuschagne'}
Failed to find the person from: {u'slug': u'cathlene-labuschagne'}
Failed to find the person from: {u'slug': u'cathlene-labuschagne'}
Failed to find the person from: {u'slug': u'hlengiwe-octavia-maxon'}
Failed to find the person from: {u'slug': u'marshall-mzingisi-dlamini'}
Failed to find the person from: {u'slug': u'marshall-mzingisi-dlamini'}
Failed to find the person from: {u'slug': u'tandi-gloria-mpambo-sibhukwana'}
Failed to find the person from: {u'slug': u'tandi-gloria-mpambo-sibhukwana'}
Failed to find the person from: {u'slug': u'tandi-gloria-mpambo-sibhukwana'}
```

Just going by the top Google hit for the old slug on www.pa.org.za, I guess the corrections should be:

    cathlene-labuschagne => cathleen-labuschagne
    hlengiwe-octavia-maxon => hlengiwe-octavia-hlophe
    tandi-gloria-mpambo-sibhukwana => thandi-gloria-mpambo-sibhukwana
    marshall-mzingisi-dlamini => mzingisi-marshall-dlamini

Thanks for pointing out the problem in #1515 again - I'm sorry that got lost. I've raised it to the top of the issues we'll consider when we plan the next sprint.

geoffkilpin commented:
Thank you @mhl for the useful comments - I've made the corrections, and fixed one of the interests browser views that I omitted the source link for. I understand @duncanparkes additions [here](https://github.com/mysociety/pombola/compare/members-interests-2015) cover the slug corrections (thank you for that as well).

mhl commented:
I've deployed this on the live site now - I added a small fix to the tests into this commit https://github.com/mysociety/pombola/commit/88d52610bd569b81461055f27f6e427e7a648519 to go with the change @duncanparkes made there.

Thanks very much, @geoffkilpin - please let us know if you spot any problems on the live site.
